### ISSUE TITLE
Generic-alike type checks also on method calls

### ISSUE THREAD

**Issue by Ocramius** [Roave/StrictPhp#11]
When calling a method annotated like following:

```php
/**
 * @param stdClass[] $foo
 */
public function doBar(array $foo) {}
```

We should intercept the call and verify the type validity of `$foo` and of all parameters

**Comment by Ocramius** [comment]
@malukenho this is about simply using the `ApplyTypeChecks` to every parameter of every public method call :D

**Comment by Ocramius** [comment]
Implemented in #20 / #23
